src/pubtools/_pulp/tasks/delete.py

```diff
@@ -564,26 +577,39 @@ def map_to_repo(self, units, repos, unit_attr):
         repo_map = {}
         unit_map = {}
         repos = sorted(repos)
-
-        for unit in sorted(units):
-            unit_name = getattr(unit, unit_attr)
-            unit_map.setdefault(unit_name, RemoveUnitItem(unit=unit, repos=[]))
-            for repo in repos:
-                if repo not in unit.repository_memberships:
-                    LOG.warning(
-                        "%s is not present in %s",
-                        unit_name,
-                        repo,
-                    )
-                else:
+        if repos:
+            for unit in sorted(units):
+                unit_name = getattr(unit, unit_attr)
+                unit_map.setdefault(unit_name, RemoveUnitItem(unit=unit, repos=[]))
+                for repo in repos:
+                    if repo not in unit.repository_memberships:
+                        LOG.warning(
+                            "%s is not present in %s",
+                            unit_name,
+                            repo,
+                        )
+                    else:
+                        repo_map.setdefault(repo, []).append(unit)
+                        unit_map.get(unit_name).repos.append(repo)
+
+            missing = set(repos) - set(repo_map.keys())
+            if missing:
+                missing = ", ".join(sorted(list(missing)))
+                LOG.warning("No units to remove from %s", missing)
+        else:
```

Comment:
Is this else branch really needed? It duplicates a few lines from the if branch. Can this somehow be combined? It is still deleting content from repos - only repos are calculated in a different way.